commit 4dadc8f75196f3aa78ed3d5c767570b7121ada19 (refs/changes/30/1030/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-11-30T14:46:33-08:00 (2 years, 10 months ago)
    
    TOOLS-1624 sdcadm crashes on platform install image not in channel

diff --git a/lib/platform.js b/lib/platform.js
index 74342ee..f9eba36 100644
--- a/lib/platform.js
+++ b/lib/platform.js
@@ -429,6 +429,7 @@ Platform.prototype.install = function install(opts, callback) {
     var self = this;
     var localdir = '/var/tmp';
     var deleteOnFinish = true;
+    var defaultChannel;
     var filepath;
     var image;
     var progress = self.progress;
@@ -588,6 +589,7 @@ Platform.prototype.install = function install(opts, callback) {
                 if (err) {
                     return next();
                 }
+                defaultChannel = channel;
                 if (!fs.existsSync(opts.image)) {
                     progress('Using channel %s', channel);
                 }
@@ -727,7 +729,7 @@ Platform.prototype.install = function install(opts, callback) {
             } else if (imgNotFoundError) {
                 progress('Unable to find the given Platform Image \'%s\' ' +
                         'in channel \'%s\'', opts.image,
-                        self.sdcadm.updates.channel);
+                        self.sdcadm.updates.channel || defaultChannel);
             } else {
                 progress('In order not to have to re-download image, ' +
                          '%s has been left behind.', filepath);
